torch/lib/THD/master_worker/common/ByteArray.cpp

```diff
@@ -11,111 +11,52 @@
 namespace thd { namespace rpc {
 
 ByteArray::ByteArray()
-  : _data(nullptr)
-  , _length(0)
-  , _size(0)
+  : _data()
 {}
 
 ByteArray::ByteArray(std::size_t size)
-  : _data(nullptr)
-  , _length(0)
-  , _size(0)
+  : ByteArray()
 {
-  _resize(size);
+  _data.reserve(size);
 }
 
-ByteArray::ByteArray(char* arr, std::size_t size)
-  : _data(arr)
-  , _length(size)
-  , _size(_length)
+ByteArray::ByteArray(const char* arr, std::size_t size)
+  : _data(arr, arr + size)
 {}
 
 ByteArray::ByteArray(ByteArray&& arr)
-  : _data(nullptr)
-  , _length(0)
-  , _size(0)
 {
   std::swap(_data, arr._data);
-  std::swap(_length, arr._length);
-  std::swap(_size, arr._size);
 }
 
 ByteArray::ByteArray(const ByteArray& arr)
-  : _data(nullptr)
-  , _length(0)
-  , _size(0)
-{
-  _resizeExact(arr._length);
-  _length = arr._length;
-  std::memcpy(_data, arr._data, _length);
-}
+  : _data(arr._data)
+{}
 
-ByteArray::~ByteArray() {
-  std::free(_data);
-}
+ByteArray::~ByteArray() {}
 
 ByteArray& ByteArray::append(const char* arr, std::size_t size) {
-  this->_resize(_length + size);
-  std::memcpy(_data + _length, arr, size);
-  _length += size;
-  return *this;
-}
-
-char* ByteArray::data() const {
-  return _data;
-}
-
-std::size_t ByteArray::length() const {
-  return _length;
-}
-
-std::string ByteArray::to_string() {
-  return std::string(_data, _length);
-}
+  if (_data.capacity() < _data.size() + size) {
```

Comment:
I don't think this is really necessary. `reserve` never makes the block smaller, and `append` should take care of resizing the string appropriately.